### ISSUE TITLE
GBK file name encoding (or Non-English file name) caused a problem

### ISSUE THREAD

**Issue by mydansun** [electron/windows-installer#94]
My computer language is Chinese, when I run electron-builder I got a problem, Log is as follows
```
Removing dist\win and dist\win-unpacked
Packaging app for platform win32 x64 using electron v1.1.1
Error: E:\UserDocument\Desktop\PROJECT\Electron\whatever\node_modules\electron-winstaller-fixed\vendor\Update.com failed with exit code: 4294967295
    at ChildProcess.proc.on.code (E:\UserDocument\Desktop\PROJECT\Electron\whatever\node_modules\electron-winstaller-fixed\lib\spawn-promise.js:33:16)
.....
```

So, I checked the SquirrelSetup, and then I found the error:

```
Unhandled exception: System.ArgumentException: 已添加项。字典中的关键字:“lib/net45/.minecraft/shaderpacks/????????V4??????.zip”所添加的关键字:“lib/net45/.minecraft/shaderpacks/????????V4??????.zip”
   在 System.Collections.Hashtable.Insert(Object key, Object nvalue, Boolean add)
```
The error told me that the file "????????V4??????.zip" has already been added, but in fact, I have two different files "真实光影V4低配" and "真实光影V4完整", because these file name are encoded to GBK rather than UTF8, their real names are missing!

how to solve it?

**Comment by develar** [comment]
> when I run electron-builder I got a problem,

It seems it it maybe electron-builder issue, not windows-installer. I will investigate.

FYI: https://github.com/atom/rcedit/issues/17

**Comment by mydansun** [comment]
but the packing function in windows-installer?

Instead, and I use temporarily [NuGetPackageExplorer](https://github.com/NuGetPackageExplorer/NuGetPackageExplorer)

**Comment by develar** [comment]
@mydansun Please try electron-builder `3.26.2`

**Comment by towertop** [comment]
@develar 

I still had this issue on electron-builder 5.1.0 and electron-winstaller-fixed 2.10.1 .

I got the same error code after I set build.win.msi as true in development package.json for electron-builder . After removing it the error gone.

```
Error: D:\Workspace\Atom\mmtrix-opttools\node_modules\electron-winstaller-fixed\vendor\Update.com failed with exit code:
 4294967295
    at ChildProcess.<anonymous> (D:\Workspace\Atom\mmtrix-opttools\node_modules\electron-winstaller-fixed\lib\spawn-prom
ise.js:34:16)
From previous event:
    at spawn (D:\Workspace\Atom\mmtrix-opttools\node_modules\electron-winstaller-fixed\lib\spawn-promise.js:23:10)
    at msi (D:\Workspace\Atom\mmtrix-opttools\node_modules\electron-winstaller-fixed\lib\index.js:319:34)
    at D:\Workspace\Atom\mmtrix-opttools\node_modules\electron-winstaller-fixed\lib\index.js:155:13
    at [object Generator].next (native)
    at ChildProcess.<anonymous> (D:\Workspace\Atom\mmtrix-opttools\node_modules\rcedit\lib\rcedit.js:54:16)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:821:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
From previous event:
    at build (D:\Workspace\Atom\mmtrix-opttools\node_modules\electron-winstaller-fixed\lib\index.js:163:16)
    at Object.<anonymous> (D:\Workspace\Atom\mmtrix-opttools\node_modules\electron-winstaller-fixed\lib\index.js:82:13)
    at [object Generator].next (native)
    at D:\Workspace\Atom\mmtrix-opttools\node_modules\fs-extra-p\node_modules\fs-extra\lib\mkdirs\mkdirs.js:35:14
    at FSReqWrap.oncomplete (fs.js:82:15)
From previous event:
    at Object.createWindowsInstaller (D:\Workspace\Atom\mmtrix-opttools\node_modules\electron-winstaller-fixed\lib\index
.js:92:16)
    at SquirrelWindowsTarget.<anonymous> (D:\Workspace\Atom\mmtrix-opttools\node_modules\electron-builder\src\targets\sq
uirrelWindows.ts:26:33)
    at [object Generator].next (native)
    at deleteItem (D:\Workspace\Atom\mmtrix-opttools\node_modules\fs-extra-p\node_modules\fs-extra\lib\empty\index.js:19
:25)
    at D:\Workspace\Atom\mmtrix-opttools\node_modules\fs-extra-p\node_modules\fs-extra\lib\empty\index.js:22:9
    at next (D:\Workspace\Atom\mmtrix-opttools\node_modules\rimraf\rimraf.js:72:7)
    at CB (D:\Workspace\Atom\mmtrix-opttools\node_modules\rimraf\rimraf.js:108:9)
From previous event:
    at tsAwaiter (D:\Workspace\Atom\mmtrix-opttools\node_modules\electron-builder\src\awaiter.ts:10:47)
    at Object.build (D:\Workspace\Atom\mmtrix-opttools\node_modules\electron-builder\src\builder.ts:176:52)
    at Object.<anonymous> (D:\Workspace\Atom\mmtrix-opttools\node_modules\electron-builder\src\build-cli.ts:8:28)
    at Module._compile (module.js:397:26)
    at Object.Module._extensions..js (module.js:404:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:429:10)
    at startup (node.js:139:18)
    at node.js:999:3
npm ERR! Windows_NT 6.3.9600
npm ERR! argv "C:\\nodejs\\node.exe" "C:\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "dist"
npm ERR! node v5.4.1
npm ERR! npm  v3.8.6
npm ERR! code ELIFECYCLE
npm ERR! @ dist: `build --win --x64`
npm ERR! Exit status 4294967295
npm ERR!
npm ERR! Failed at the @ dist script 'build --win --x64'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the  package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     build --win --x64
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     D:\Workspace\Atom\mmtrix-opttools\npm-debug.log
```

**Comment by develar** [comment]
@towertop https://github.com/electron-userland/electron-builder/issues/485
